packages/kit/src/core/sync/write_types/index.js

```diff
@@ -465,6 +465,12 @@ function process_node(node, outdir, is_page, proxies, all_pages_have_load = true
 
 	exports.push(`export type ${prefix}Data = ${data};`);
 
+	exports.push(
+		`export type ${prefix}Props = { data: ${prefix}Data; ${
+			node.server && is_page ? 'form: ActionData; ' : ''
```

Comment:
Pretty dumb way to write this I know. I could've pushed this in the nested if above but I just thought writing it below `PageData` would be more intuitive. Also, didn't know how to add `children` in `LayoutProps` since I can't write `import("svelte").Snippet` as svelte isn't included.